lib/rules/no-control-regex.js

```diff
@@ -51,29 +51,47 @@ module.exports = {
             return null;
         }
 
+
+        const controlChar = /[\x00-\x1f]/g; // eslint-disable-line no-control-regex
+        const multipleSlashes = /\\+/g;
+        const multipleSlashesAtEnd = /\\+$/g;
+
         /**
-         * Check if given regex string has control characters in it
+         * Return a list of the control characters in the given regex string
          * @param {string} regexStr regex as string to check
-         * @returns {boolean} returns true if finds control characters on given string
+         * @returns {array} returns a list of found control characters on given string
          * @private
          */
-        function hasControlCharacters(regexStr) {
+        function getControlCharacters(regexStr) {
 
             // check control characters, if RegExp object used
-            let hasControlChars = /[\x00-\x1f]/.test(regexStr); // eslint-disable-line no-control-regex
+            const controlChars = regexStr.match(controlChar) || [];
+
+            let stringControlChars = [];
 
             // check substr, if regex literal used
             const subStrIndex = regexStr.search(/\\x[01][0-9a-f]/i);
 
-            if (!hasControlChars && subStrIndex > -1) {
+            if (subStrIndex > -1) {
 
                 // is it escaped, check backslash count
-                const possibleEscapeCharacters = regexStr.substr(0, subStrIndex).match(/\\+$/gi);
+                const possibleEscapeCharacters = regexStr.slice(0, subStrIndex).match(multipleSlashesAtEnd);
 
-                hasControlChars = possibleEscapeCharacters === null || !(possibleEscapeCharacters[0].length % 2);
+                const hasControlChars = possibleEscapeCharacters === null || !(possibleEscapeCharacters[0].length % 2);
```

Comment:
This is where we check for an even number of slashes - note that this check happens in master, I'm just moving the regex higher up.